src/awkward/_prettyprint.py

```diff
@@ -35,38 +34,6 @@ def alternate(length):
 is_identifier = re.compile(r"^[A-Za-z_][A-Za-z_0-9]*$")
 
 
-# avoid recursion in which ak.Array.__getitem__ calls prettyprint
-# to form an error string: private reimplementation of ak.Array.__getitem__
-
-
-def get_at(data, index):
-    out = data._layout._getitem_at(index)
-    if isinstance(out, ak.contents.NumpyArray):
-        array_param = out.parameter("__array__")
-        if array_param == "byte":
-            return ak._util.tobytes(out._raw(numpy))
-        elif array_param == "char":
-            return ak._util.tobytes(out._raw(numpy)).decode(errors="surrogateescape")
-    if isinstance(out, (ak.contents.Content, ak.record.Record)):
-        return wrap_layout(out, data._behavior)
-    else:
-        return out
-
-
-def get_field(data, field):
-    out = data._layout._getitem_field(field)
-    if isinstance(out, ak.contents.NumpyArray):
-        array_param = out.parameter("__array__")
-        if array_param == "byte":
-            return ak._util.tobytes(out._raw(numpy))
-        elif array_param == "char":
-            return ak._util.tobytes(out._raw(numpy)).decode(errors="surrogateescape")
-    if isinstance(out, (ak.contents.Content, ak.record.Record)):
-        return wrap_layout(out, data._behavior)
-    else:
-        return out
```

Comment:
Why is it okay to remove these? I think the problem they were addressing is that there was an error in constructing the error message, which is not easily tested in automated tests.

Also, they avoid the rather complex dispatch of `Content._getitem`, which checks for any kind of slice, with `Content._getitem_at` or `Content._getitem_field`. That's not (just) a performance thing, but it also avoids the possibility of errors in that dispatch, which can be a problem if  it needs to print the array when there's an error.